### PR TITLE
Fixing shipping address override in nxo flow

### PR DESCRIPTION
**Pull request by kelay858**
Thank you for your contribution to Braintree. 

> Before submitting this PR, note that we cannot accept language translation PRs. We support the same languages that are supported by PayPal, and have a dedicated localization team to provide the translations. If there is an error in a specific translation, you may open an issue and we will escalate it to the localization team.

### Summary of changes

 - Updating the native version and including return url
 - Fixing a bug with address override not being set

 ### Checklist

 - [x] Added a changelog entry

-

### CODE REVIEW

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutVaultRequest.java
Comment (jaxdesmarais) [inline PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutVaultRequest.java:91]: Is this a breaking change? Java is far from my forte so not sure if removing `PayPalNativeCheckoutVaultRequest` from the `Creator` is considered a breaking change in java. If it is, we should call it out in the changelog.

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java
Comment (jaxdesmarais) [inline PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java:83]: I think we can use a similar docstring here to what we use in standard PayPal request: https://github.com/braintree/braintree_android/blob/b2ced4e6bc983a10a5ae7d8eebddcb8808f97c7d/PayPal/src/main/java/com/braintreepayments/api/PayPalRequest.java#L89-L96

Comment (jaxdesmarais) [reply PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java:83]: Ah, but that string is for the setter. Do we have a setter for this? In any case I think we can mimic the docstrings there for all of these properties (same with the ones below)

File: CHANGELOG.md
Comment (jaxdesmarais) [inline CHANGELOG.md:10]: General question, is this a different return URL than the one that was removed before.

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutVaultRequest.java
Comment (kelay858) [reply PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutVaultRequest.java:91]: nah java will cast to the type declaration

File: CHANGELOG.md
Comment (kelay858) [reply CHANGELOG.md:10]: no its the same one, its a requirement on oath 2.0 to have this in so we had to add it back to native

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java
Comment (jaxdesmarais) [reply PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java:83]: Ah, no I was confused - I didn't expand things and see the setter above with the same docstring as the docs I linked. 🤦

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutVaultRequest.java
Comment (jaxdesmarais) [reply PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutVaultRequest.java:91]: Nice! That's dope.

File: CHANGELOG.md
Comment (jaxdesmarais) [reply CHANGELOG.md:10]: Dang, that's a bummer to have to add it back in. Do you know if we should expect a similar PR for iOS? We should also add this to the docs. Just curious what the experience will be if a merchant doesn't include this? Is this auth change already live or waiting until we do a release to make updates?

Comment (kelay858) [reply CHANGELOG.md:10]: Yup we need to make a small change to the dev docs to add this in! I can work on doing that, is there a process for that? I remember there being a git somewhere. IOS will not need this change, its an android only requirement since their SVC can just close without context. There is actually no change on auth here, its all client side with the connect flow. If a merchant does not include this, they will be stuck after auth and will not be redirected to their application, which is expected as thats the current behavior on nxo

Comment (jaxdesmarais) [reply CHANGELOG.md:10]: Yeah the [docs repo is here](https://github.braintreeps.com/braintree/docs) - you'd need to open up a PR to it with this change, once it's merged it gets auto released by the docs team weekly.

And thanks for clarifying that it's only needed on Android!

Comment (jaxdesmarais) [inline CHANGELOG.md:10]:
```suggestion
  * Breaking change
    *`PayPalNativeRequest` requires a `returnUrl` to redirect correctly after authentication
```
Thinking we should maybe call this out as a breaking change

**Review by jaxdesmarais — APPROVED** [review]

File: CHANGELOG.md
Comment (sshropshire) [reply CHANGELOG.md:10]: +1 on the breaking change callout.

**Review by sshropshire — APPROVED** [review]

File: CHANGELOG.md
Comment (kelay858) [reply CHANGELOG.md]: thank you! Will open a PR now

**Pull request merged**
